### PR TITLE
Testing all models for forward

### PR DESCRIPTION
**Pull request by nicolas-chaulet**
This PR is mainly about running a foward and backward on all of our models.

@humanpose1 I have modified your registration models slightly, because they were failing to forward in the tests (probably invalid kind of data). Please double check that I haven't broken anything!

### CODE REVIEW

File: conf/models/registration/pointnet2.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@ models:
       radii: [[0.2], [0.4]]
       nsamples: [[64], [64]]
       down_conv_nn:
-        [[[FEAT, 32, 32, 64]], [[64+3, 64, 128, 128]]]
+        [[[FEAT +3, 32, 32, 64]], [[64+3, 64, 128, 128]]]
```
Comment (humanpose1) [inline conf/models/registration/pointnet2.yaml]: The problem here is that the feature here is only XYZ. I got an error when data.x was empty...

Comment (nicolas-chaulet) [reply conf/models/registration/pointnet2.yaml]: It shouldn't anymore, let me know if you still see it

File: src/models/registration/pointnet2.py [hunk 2]
```diff
@@ -91,4 +91,5 @@ def backward(self):
         """Calculate losses, gradients, and update network weights; called in every training iteration"""
         # caculate the intermediate results if necessary; here self.output has been computed during function <forward>
         # calculate loss given the input and intermediate results
-        self.loss.backward()  # calculate gradients of network G w.r.t. loss_G
+        if hasattr(self, "loss") and self.loss:
```
Comment (tchaton) [inline src/models/registration/pointnet2.py]: Are you sure you can do that ? Won't it trigger an error if you don't have loss in self.

File: conf/models/registration/pointnet2.yaml (hunk 1 above)
Comment (humanpose1) [reply conf/models/registration/pointnet2.yaml]: I know why you had this problem. I add XYZ in the transforms whereas you add in the network(I put use_xyz=False)

File: test/mockdatasets.py [hunk 3]
```diff
@@ -18,10 +18,10 @@ def get(self, dataset_name, default):
 
 
 class MockDataset(torch.utils.data.Dataset):
-    def __init__(self, feature_size=0):
+    def __init__(self, feature_size=0, transform=None, num_points=100):
```
Comment (tchaton) [inline test/mockdatasets.py]: I would prefer to have input_nc, output_nc there. Because by default, this mock dataset works only for segmentation. Would be great to have as generic as possible.

File: test/test_base_dataset.py [hunk 4]
```diff
@@ -1,149 +0,0 @@
...
-        self.assertEqual(dataset.test_dataset, None)
-        self.assertEqual(dataset.val_dataset, None)
-
-    def test_simple_datasets(self):
-
-        opt = Options()
-        opt.dataset_name = os.path.join(os.getcwd(), "test")
-        opt.dataroot = os.path.join(os.getcwd(), "test")
-        
-        class SimpleDataset(BaseDataset):
```
Comment (tchaton) [inline test/test_base_dataset.py]: I liked using directly BaseDataset for Mocking the dataset. I think it would bring more robustness within the code.

File: test/test_models.py [hunk 5]
```diff
@@ -41,26 +42,46 @@ def test_createall(self):
...
-                    models_config.update("model_name", model_name)
-                    instantiate_model(models_config, MockDatasetGeometric(6))
+                with self.subTest(model_name):
+                    if model_name not in ["MyTemplateModel", "MinkUNet_WIP"]:
+                        models_config.update("model_name", model_name)
+                        instantiate_model(models_config, MockDatasetGeometric(6))
+
+    def test_runall(self):
+        def is_known_to_fail(model_name):
+            forward_failing = ["MyTemplateModel", "MinkUNet_WIP", "pointcnn", "RSConv_4LD", "RSConv_2LD", "randlanet"]
```
Comment (tchaton) [inline test/test_models.py]: MyTemplateModel doesn't t exist anymore I think.

File: test/test_base_dataset.py (hunk 4 above)
Comment (nicolas-chaulet) [reply test/test_base_dataset.py]: I haven't deleted your code, just moved it to a different place

File: test/test_models.py (hunk 5 above)
Comment (nicolas-chaulet) [reply test/test_models.py]: ok cool

File: src/models/registration/pointnet2.py (hunk 2 above)
Comment (humanpose1) [reply src/models/registration/pointnet2.py]: You can see[here](https://github.com/nicolas-chaulet/deeppointcloud-benchmarks/blob/74784e2b37f2c7c24bc819eee9deef00a4152dda/src/models/base_architectures/backbone.py#L101), where we add the loss. If in the config, no loss has been provided, it will be None. You're right. We should throw an exception if the loss is None (The miner can be None)

File: test/mockdatasets.py (hunk 3 above)
Comment (nicolas-chaulet) [reply test/mockdatasets.py]: Agreed, let's do that when we need to use it for another task

File: src/models/registration/pointnet2.py (hunk 2 above)
Comment (nicolas-chaulet) [reply src/models/registration/pointnet2.py]: Nope, it is tested :)

Comment (nicolas-chaulet) [reply src/models/registration/pointnet2.py]: The only way the loss does not exists is if the forward functions stops earlier (there is a return half way through). I'll leave the hasattr but remove the second check so that it throws an exception
